pyproject.toml

```diff
@@ -167,6 +167,27 @@ extend_skip_glob = ["asdf/extern/*", ".eggs/*", ".tox/*"]
 
 [tool.ruff]
 line-length = 120
+select = ["ALL"]
+extend-ignore = [
+    # Ignored check groups
+    "C90", # mccabe complexity
+    "D", # pydocstyle
+    "ANN", # flake8-annotations
+    "FBT", # flake8-boolean-trap
```

Comment:
This check basically points out that having boolean defaults for methods can lead to weird problems. Something like an `Enum` is generally a better practice as it allows us to expand the possible options. I wonder if we should consider addressing the errors this check points out.

Reply:
The errors found by this are:
```shell
asdf/asdf.py:167:34: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:385:17: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:581:27: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:746:30: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:751:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/asdf.py:770:34: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:805:30: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:852:31: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:960:21: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:964:21: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:981:30: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:1076:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/asdf.py:1183:34: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:1187:21: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:1203:31: EM101 Exception must not use a string literal, assign to variable first
asdf/asdf.py:1680:26: EM102 Exception must not use an f-string literal, assign to variable first
asdf/asdf.py:1712:29: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:74:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:77:30: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:93:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:118:30: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:210:34: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:353:34: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:626:34: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:632:34: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:654:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:666:29: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:693:38: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:700:26: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:1006:21: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:1020:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:1035:30: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:1038:30: EM101 Exception must not use a string literal, assign to variable first
asdf/block.py:1086:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:1150:32: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:1185:40: EM102 Exception must not use an f-string literal, assign to variable first
asdf/block.py:1195:31: EM101 Exception must not use a string literal, assign to variable first
asdf/commands/diff.py:138:29: EM101 Exception must not use a string literal, assign to variable first
asdf/commands/edit.py:328:26: EM101 Exception must not use a string literal, assign to variable first
asdf/commands/extract.py:46:36: EM102 Exception must not use an f-string literal, assign to variable first
asdf/commands/tests/test_edit.py:99:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/compression.py:49:26: EM102 Exception must not use an f-string literal, assign to variable first
asdf/compression.py:60:17: EM101 Exception must not use a string literal, assign to variable first
asdf/compression.py:214:26: EM102 Exception must not use an f-string literal, assign to variable first
asdf/compression.py:271:26: EM101 Exception must not use a string literal, assign to variable first
asdf/config.py:93:30: EM101 Exception must not use a string literal, assign to variable first
asdf/config.py:177:30: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_converter.py:162:29: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_converter.py:177:33: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_converter.py:181:17: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_converter.py:191:33: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_extension.py:137:29: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_extension.py:156:33: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_extension.py:167:29: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_extension.py:176:33: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_extension.py:189:25: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_manager.py:129:17: EM102 Exception must not use an f-string literal, assign to variable first
asdf/extension/_manager.py:154:17: EM102 Exception must not use an f-string literal, assign to variable first
asdf/extension/_manager.py:182:21: EM102 Exception must not use an f-string literal, assign to variable first
asdf/extension/_manifest.py:112:33: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_tag.py:25:30: EM101 Exception must not use a string literal, assign to variable first
asdf/extension/_tag.py:73:32: EM101 Exception must not use a string literal, assign to variable first
asdf/fits_embed.py:14:23: EM101 Exception must not use a string literal, assign to variable first
asdf/fits_embed.py:73:34: EM102 Exception must not use an f-string literal, assign to variable first
asdf/fits_embed.py:85:30: EM101 Exception must not use a string literal, assign to variable first
asdf/fits_embed.py:255:34: EM102 Exception must not use an f-string literal, assign to variable first
asdf/fits_embed.py:303:30: EM101 Exception must not use a string literal, assign to variable first
asdf/fits_embed.py:342:35: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:71:26: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:142:46: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:153:42: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:206:30: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:243:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:317:30: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:339:32: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:611:35: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:617:35: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:624:35: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:741:34: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:855:27: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:1007:26: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:1014:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:1021:34: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:1042:25: EM101 Exception must not use a string literal, assign to variable first
asdf/generic_io.py:1046:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:1063:34: EM102 Exception must not use an f-string literal, assign to variable first
asdf/generic_io.py:1082:22: EM102 Exception must not use an f-string literal, assign to variable first
asdf/reference.py:39:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/reference.py:89:34: EM102 Exception must not use an f-string literal, assign to variable first
asdf/reference.py:167:26: EM101 Exception must not use a string literal, assign to variable first
asdf/resolver.py:50:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/resolver.py:86:34: EM102 Exception must not use an f-string literal, assign to variable first
asdf/resource.py:45:29: EM101 Exception must not use a string literal, assign to variable first
asdf/resource.py:147:28: EM102 Exception must not use an f-string literal, assign to variable first
asdf/schema.py:335:33: EM102 Exception must not use an f-string literal, assign to variable first
asdf/schema.py:604:35: EM102 Exception must not use an f-string literal, assign to variable first
asdf/schema.py:633:39: EM102 Exception must not use an f-string literal, assign to variable first
asdf/search.py:157:29: EM101 Exception must not use a string literal, assign to variable first
asdf/search.py:244:32: EM101 Exception must not use a string literal, assign to variable first
asdf/search.py:262:32: EM101 Exception must not use a string literal, assign to variable first
asdf/search.py:347:29: EM101 Exception must not use a string literal, assign to variable first
asdf/search.py:408:30: EM101 Exception must not use a string literal, assign to variable first
asdf/tagged.py:141:29: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/integer.py:53:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/integer.py:112:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/ndarray.py:36:22: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/ndarray.py:59:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/ndarray.py:77:36: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:79:22: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/ndarray.py:129:22: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/ndarray.py:146:34: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:231:34: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:307:34: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:311:26: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/ndarray.py:364:34: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tags/core/ndarray.py:377:34: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:389:25: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:432:21: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:495:38: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:658:35: EM101 Exception must not use a string literal, assign to variable first
asdf/tags/core/ndarray.py:662:31: EM101 Exception must not use a string literal, assign to variable first
asdf/tests/helpers.py:486:21: EM102 Exception must not use an f-string literal, assign to variable first
asdf/tests/test_entry_points.py:42:21: EM101 Exception must not use a string literal, assign to variable first
asdf/tests/test_entry_points.py:106:21: EM101 Exception must not use a string literal, assign to variable first
asdf/tests/test_helpers.py:18:29: EM101 Exception must not use a string literal, assign to variable first
asdf/tests/test_info.py:82:38: EM101 Exception must not use a string literal, assign to variable first
asdf/tests/test_info.py:86:38: EM101 Exception must not use a string literal, assign to variable first
asdf/tests/test_schema.py:1070:17: EM103 Exception must not use a `.format()` string directly, assign to variable first
asdf/tests/test_versioning.py:363:13: EM102 Exception must not use an f-string literal, assign to variable first
asdf/treeutil.py:127:17: EM101 Exception must not use a string literal, assign to variable first
asdf/treeutil.py:188:32: EM101 Exception must not use a string literal, assign to variable first
asdf/treeutil.py:268:26: EM101 Exception must not use a string literal, assign to variable first
asdf/type_index.py:56:30: EM102 Exception must not use an f-string literal, assign to variable first
asdf/type_index.py:225:29: EM101 Exception must not use a string literal, assign to variable first
asdf/types.py:41:21: EM102 Exception must not use an f-string literal, assign to variable first
asdf/types.py:113:33: EM101 Exception must not use a string literal, assign to variable first
asdf/types.py:136:29: EM101 Exception must not use a string literal, assign to variable first
asdf/util.py:198:32: EM102 Exception must not use an f-string literal, assign to variable first
asdf/util.py:227:32: EM102 Exception must not use an f-string literal, assign to variable first
asdf/util.py:373:13: EM102 Exception must not use an f-string literal, assign to variable first
asdf/yamlutil.py:179:17: EM101 Exception must not use a string literal, assign to variable first
asdf/yamlutil.py:187:21: EM101 Exception must not use a string literal, assign to variable first
asdf/yamlutil.py:194:21: EM101 Exception must not use a string literal, assign to variable first
asdf/yamlutil.py:241:29: EM102 Exception must not use an f-string literal, assign to variable first
asdf/yamlutil.py:383:25: EM101 Exception must not use a string literal, assign to variable first
compatibility_tests/common.py:10:13: EM103 Exception must not use a `.format()` string directly, assign to variable first
pytest_asdf/plugin.py:177:32: EM101 Exception must not use a string literal, assign to variable first
```